packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -590,3 +608,179 @@ class ThemeData {
+///
+/// The localization is done by merging of the [TextTheme] fields of the
+/// [delegate] into the [localTextGeometry] and caching the results.
+// Suppressing @immutable warning because this class contains lazily computed
+// fields. Otherwise, for all intents and purposes, this class is immutable.
+// ignore: must_be_immutable
+class _LocalizedThemeData implements ThemeData {
+  _LocalizedThemeData(this.delegate, this.localTextGeometry);
+
+  final ThemeData delegate;
+  final TextTheme localTextGeometry;
+
+  @override
+  Color get accentColor => delegate.accentColor;
+
+  @override
+  Brightness get accentColorBrightness => delegate.accentColorBrightness;
+
+  @override
+  IconThemeData get accentIconTheme => delegate.accentIconTheme;
+
+  @override
+  Color get backgroundColor => delegate.backgroundColor;
+
+  @override
+  Brightness get brightness => delegate.brightness;
+
+  @override
+  Color get buttonColor => delegate.buttonColor;
+
+  @override
+  Color get canvasColor => delegate.canvasColor;
+
+  @override
+  Color get cardColor => delegate.cardColor;
+
+  @override
+  Color get dialogBackgroundColor => delegate.dialogBackgroundColor;
+
+  @override
+  Color get disabledColor => delegate.disabledColor;
+
+  @override
+  Color get dividerColor => delegate.dividerColor;
+
+  @override
+  Color get errorColor => delegate.errorColor;
+
+  @override
+  Color get highlightColor => delegate.highlightColor;
+
+  @override
+  Color get hintColor => delegate.hintColor;
+
+  @override
+  IconThemeData get iconTheme => delegate.iconTheme;
+
+  @override
+  Color get indicatorColor => delegate.indicatorColor;
+
+  @override
+  TargetPlatform get platform => delegate.platform;
+
+  @override
+  Color get primaryColor => delegate.primaryColor;
+
+  @override
+  Brightness get primaryColorBrightness => delegate.primaryColorBrightness;
+
+  @override
+  IconThemeData get primaryIconTheme => delegate.primaryIconTheme;
+
+  @override
+  Color get scaffoldBackgroundColor => delegate.scaffoldBackgroundColor;
+
+  @override
+  Color get secondaryHeaderColor => delegate.secondaryHeaderColor;
+
+  @override
+  Color get selectedRowColor => delegate.selectedRowColor;
+
+  @override
+  Color get splashColor => delegate.splashColor;
+
+  @override
+  Color get textSelectionColor => delegate.textSelectionColor;
+
+  @override
+  Color get textSelectionHandleColor => delegate.textSelectionHandleColor;
+
+  @override
+  Color get unselectedWidgetColor => delegate.unselectedWidgetColor;
+
+  @override
+  TextTheme get primaryTextTheme => _primaryTextTheme ??= delegate.primaryTextTheme.merge(localTextGeometry);
+  TextTheme _primaryTextTheme;
+
+  @override
+  TextTheme get accentTextTheme => _accentTextTheme ??= delegate.accentTextTheme.merge(localTextGeometry);
+  TextTheme _accentTextTheme;
+
+  @override
+  TextTheme get textTheme => _textTheme ??= delegate.textTheme.merge(localTextGeometry);
+  TextTheme _textTheme;
+
+  @override
+  ThemeData copyWith({
```

Comment:
how does this differ from the superclass implementation? if it does, please add a comment saying what the difference is